packages/react-router-dev/vite/plugin.ts

```diff
@@ -738,6 +738,17 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = () => {
         viteConfigEnv = _viteConfigEnv;
         viteCommand = viteConfigEnv.command;
 
+        let viteClientConditions: string[] =
+          viteUserConfig.resolve?.conditions ??
+          "defaultClientConditions" in vite
+            ? Array.from(vite.defaultClientConditions)
+            : [];
+        let viteServerConditions: string[] =
+          viteUserConfig.ssr?.resolve?.conditions ??
+          "defaultServerConditions" in vite
+            ? Array.from(vite.defaultServerConditions)
+            : [];
```

Comment:
We could make this more type safe by specifying the literal types accepted here.

```js
type DevProdCondition = 'development' | 'production';
type BaseCondition = 'module' | DevProdCondition;

type ClientCondition = BaseCondition | 'browser';
type ServerCondition = BaseCondition | 'node' | 'workerd' | 'worker';
```

You could then use this as `ClientCondition[]` or `ServerCondition[]`.

Reading the vite docs and seeing what we have here, I think this should be fine, also helps make sure we don't mistype anything.

Reply:
I don't see the benefit of coercing unknown strings to known strings.